framework/src/actions/MaterialOutputAction.C

```diff
@@ -254,45 +214,139 @@ MaterialOutputAction::materialOutput(const std::string & property_name,
+                         material,
+                         get_names_only);
 
   else if (hasProperty<RankFourTensor>(property_name))
-    names = materialOutputHelper<RankFourTensor>(property_name, material, get_names_only);
-
+    names = outputHelper({"MaterialRankFourTensorAux", "012", {"i", "j", "k", "l"}},
+                         property_name,
+                         property_name + "_",
+                         material,
+                         get_names_only);
+
+  else if (hasProperty<SymmetricRankTwoTensor>(property_name))
+    names = outputHelper({"MaterialSymmetricRankTwoTensorAux", "012345", {"component"}},
+                         property_name,
+                         property_name + "_",
+                         material,
+                         get_names_only);
+
+  else if (hasProperty<SymmetricRankFourTensor>(property_name))
+    names = outputHelper({"MaterialSymmetricRankFourTensorAux", "012345", {"i", "j"}},
+                         property_name,
+                         property_name + "_",
+                         material,
+                         get_names_only);
+
+  // Functors
   else if (hasFunctorProperty<Real>(property_name))
-    names = functorMaterialOutputHelper<Real>(property_name, material, get_names_only);
+    names = outputHelper({"FunctorMaterialRealAux", "", {}},
+                         property_name,
+                         property_name + "_out",
+                         material,
+                         get_names_only);
 
   else if (hasFunctorProperty<ADReal>(property_name))
-    names = functorMaterialOutputHelper<Real>(property_name, material, get_names_only);
+    names = outputHelper({"ADFunctorMaterialRealAux", "", {}},
+                         property_name,
+                         property_name + "_out",
+                         material,
+                         get_names_only);
 
   else if (hasFunctorProperty<RealVectorValue>(property_name))
-    names = functorMaterialOutputHelper<RealVectorValue>(property_name, material, get_names_only);
+    names = outputHelper({"FunctorMaterialRealVectorValueAux", "xyz", {"component"}},
+                         property_name,
+                         property_name + "_out_",
+                         material,
+                         get_names_only);
 
   else if (hasFunctorProperty<ADRealVectorValue>(property_name))
-    names = functorMaterialOutputHelper<RealVectorValue>(property_name, material, get_names_only);
+    names = outputHelper({"ADFunctorMaterialRealVectorValueAux", "xyz", {"component"}},
+                         property_name,
+                         property_name + "_out_",
+                         material,
+                         get_names_only);
 
   return names;
 }
 
+std::vector<std::string>
+MaterialOutputAction::outputHelper(const MaterialOutputAction::OutputMetaData & metadata,
+                                   const MaterialPropertyName & property_name,
```

Comment:
```suggestion
                                   const std::string & property_name,
```
functors can be things other than material properties so might as well keep the most generic string type